tsdb/engine/tsm1/engine.go

```diff
@@ -228,7 +258,8 @@ func (e *DevEngine) Begin(writable bool) (tsdb.Tx, error) {
 
 func (e *DevEngine) WriteTo(w io.Writer) (n int64, err error) { panic("not implemented") }
 
-func (e *DevEngine) writeSnapshot() error {
+// WriteSnapshot will snapshot the cache and write a new TSM file with its contents, releasing the snapshot when done.
+func (e *DevEngine) WriteSnapshot(async bool) error {
```

Comment:
I'm fine with ripping out the goroutine until we have more performance data. However, there are cases where I'd argue we should be panicing the server. Particularly in these early stages while we're just testing.

We don't want the server continually running and taking in writes despite the fact that things aren't working under the hood. A panic that sets off alarms is something that system operators can look into. Error messages that go into logs may or may not be checked on within any decent time frame.

We've seen this come up time and again over the last 2 years. People complained when the system would continue to take writes despite the fact that it was so overloaded that it wouldn't recover.

If your system runs out of HD space, for instance. A panic is appropriate. It will require a system operator to fix and you want to have alarms going off, which is something that a processing going down will usually trigger.